### PR TITLE
Feature residence validation

### PR DESCRIPTION
**Pull request by tramuntanal**
This PR integrates the app verification mechanism with the INE residence validation webservice.
It provides a new `consell_mallorca_authorization_handler` authorization.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,5 +20,31 @@ user.save!
...
-# decidim-mallorca-app
+
+## Residence verification
+
+A new `consell_mallorca_authorization_handler` authorization is provided for Organizations to enable it.
+In order for the residence verification integration to work, some things must be configured at different levels.
+
+### Installation configuration
+
+- The installation must have the certificates correctly placed. This is, for the production envitonment, CA certificates are expected at `config/certs/key.pem` and `config/certs/cert.pem`.
```
Comment (LauraJaimeC) [inline README.md:31]:
```suggestion
- The installation must have the certificates correctly placed. This is, for the production environment, CA certificates are expected at `config/certs/key.pem` and `config/certs/cert.pem`.
```

File: app/services/residence_verification/client.rb [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+# Connects to the "Servicio de Verificación de Datos de Residencia en un Ámbito Territorial" service
+# to check for residence registration
+module ResidenceVerification
+  class Client
+
+    attr_reader :codigo_estado_respuesta
+
+    def initialize
+      @client = Savon.client(wsdl: Rails.root.join('lib/consell_mallorca/authorization/ws/INE.VerificacionAmbitoResidencia.wsdl'), endpoint: endpoint_url) do
+      # @client = Savon.client(endpoint: endpoint_url, namespace: "http://intermediacion.redsara.es/scsp/wsdl") do
```
Comment (LauraJaimeC) [inline app/services/residence_verification/client.rb:12]: is it neccesary?

File: app/services/residence_verification/client.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      response.body
+    end
+
+    #------------------------------------------------------
+    private
+    #------------------------------------------------------
+
+    def endpoint_url
+      if Rails.env.production?
+        "https://intermediacion.redsara.es/servicios/SVD/INE.VerificacionAmbitoResidencia"
```
Comment (LauraJaimeC) [inline app/services/residence_verification/client.rb:34]: is that ok? I mean, maybe is a better practice define the values in `application.yml`. What do you think?

File: config/locales/es.yml [hunk 4]
```diff
@@ -1,10 +1,41 @@
...
+        update:
+          success: Se ha guardado correctamente el código de municipio para integrar con el sistema del padrón
+  decidim:
+    consell_mallorca_authorization:
+      admin:
+        menu: Censo
+      errors:
+        messages:
+          not_censed: No formeu part del cens
+          rq_failed: La connexió amb el cens ha fallat
```
Comment (LauraJaimeC) [inline config/locales/es.yml:30]: This translations must be in spanish :detective:

File: config/locales/es.yml [hunk 5]
```diff
@@ -1,10 +1,41 @@
...
+    authorization_handlers:
+      consell_mallorca_authorization_handler:
+        document_types:
+          nie: NIE
+          nif: NIF
+          passport: Pasaporte
+        explanation: Verificate mediante el censo del Consell de Mallorca
```
Comment (LauraJaimeC) [inline config/locales/es.yml:37]:
```suggestion
        explanation: Verifícate mediante el censo del Consell de Mallorca
```

File: spec/controllers/consell_mallorca/admin/authorization_configs_controller_spec.rb [hunk 6]
```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+
+# require Rails.root.join "app/controllers", "consell_mallorca/admin/authorization_configs_controller"
```
Comment (LauraJaimeC) [inline spec/controllers/consell_mallorca/admin/authorization_configs_controller_spec.rb:5]: it is neccesary?

File: spec/controllers/consell_mallorca/admin/authorization_configs_controller_spec.rb [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+               type: :controller do
+  include Warden::Test::Helpers
+
+  let(:organization) do
+    FactoryBot.create :organization,
+                      available_authorizations: ["consell_mallorca_census"]
+  end
+
+  let(:user) do
+    FactoryBot.create :user, :confirmed, :admin_terms_accepted, organization: organization, admin: true#, nickname: "nickname"
```
Comment (LauraJaimeC) [inline spec/controllers/consell_mallorca/admin/authorization_configs_controller_spec.rb:17]: nickname is required here?

**Comment by LauraJaimeC** [comment]
@tramuntanal  there is a test that still failing:

Captura de pantalla de 2021-10-19 16-08-35

**Comment by tramuntanal** [comment]
Thanks for this great review @LauraJaimeC . We don't want to merge because we're waiting for the customer to provide values for many fields in `ResidenceVerification::Rq::DatosGenericos`.

**Comment by tramuntanal** [comment]
Superseded by https://github.com/CodiTramuntana/decidim-mallorca-app/pull/37

**Pull request closed without merging**